From a15ef19a8ca2ed1e3d652459a082dd70c2c351e0 Mon Sep 17 00:00:00 2001
From: Viktor Szakats <commit@vsz.me>
Date: Mon, 24 Jul 2023 12:42:56 +0000
Subject: [PATCH 2/2] cmake: update ngtcp2 detection

Replace `OpenSSL` with `quictls` to follow the same change
in the v0.17.0 ngtcp2 release.

Follow-up to e0093b4b732f6495b0fb1cd6747cbfedcdcf63ed

Closes #11508
---
 CMake/FindNGTCP2.cmake | 2 +-
 CMakeLists.txt         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMake/FindNGTCP2.cmake b/CMake/FindNGTCP2.cmake
index f4e3876083737..ae92e417a5ca0 100644
--- a/CMake/FindNGTCP2.cmake
+++ b/CMake/FindNGTCP2.cmake
@@ -71,7 +71,7 @@ endif()
 if(NGTCP2_FIND_COMPONENTS)
   set(NGTCP2_CRYPTO_BACKEND "")
   foreach(component IN LISTS NGTCP2_FIND_COMPONENTS)
-    if(component MATCHES "^(BoringSSL|OpenSSL|wolfSSL|GnuTLS)")
+    if(component MATCHES "^(BoringSSL|quictls|wolfSSL|GnuTLS)")
       if(NGTCP2_CRYPTO_BACKEND)
         message(FATAL_ERROR "NGTCP2: Only one crypto library can be selected")
       endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
index babc500867bc8..7a4c36fac9d27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -604,7 +604,7 @@ if(USE_NGTCP2)
     elseif(HAVE_BORINGSSL)
       find_package(NGTCP2 REQUIRED BoringSSL)
     else()
-      find_package(NGTCP2 REQUIRED OpenSSL)
+      find_package(NGTCP2 REQUIRED quictls)
     endif()
     CheckQuicSupportInOpenSSL()
   elseif(USE_GNUTLS)
   